### PR TITLE
feat(website): give every changelog page its own social card

### PR DESCRIPTION
**Pull request by benvinegar**
Follow-up to #783. Every changelog page shared the site-wide `og.png`, with alt text describing the docs — so a shared release link showed a generic card with the version only in the small title line. Titles and descriptions were already per-page; only the image was not.

## What it looks like

Direction A of [the mockups](https://sideshow.sh/modem/benvinegar/p/Ali8aVuHUdQ) — the site's own paper surface, so a card and the page it links to read as one thing. 20 cards at 1200×630, **652KB total (~30KB each)**, one more per minor release. The preview deploy serves them at `/changelog/og/<slug>.png`.

Two content rules shape the layout:

- **A series with no editorial summary drops the tagline** rather than padding it with the factual fallback ("Release notes for Hunk 0.2: 1 release…"). About half the series have never had one written, which is why the *version* rather than the tagline is the largest element.
- **Patch chips appear only when there is more than one release to show**, where they make the one-page-per-series idea legible at a glance.

Verified against all four content shapes: current release (tagline + `LATEST` pill), superseded series (the fullest card — 4-line tagline *and* chips), a series with no summary, and the index.

## How it fits together

The changelog generator derives each card's content into `website/releases/cards.json`, so the facts on a card come from the same derivation as the page it belongs to; the renderer only decides how they look. Each page's frontmatter overrides `og:image`, `og:image:alt`, and `twitter:image` — Starlight merges page head entries over the global ones, so exactly one image tag is emitted and ordinary docs pages keep the site-wide one.

`bun run generate:og` paints the cards with Chromium. Rendering needs a browser, so it stays a maintainer step rather than part of the website build: `generate:changelog` records which cards should exist and reports missing images, **failing under `--check`**, so a new release cannot ship a page whose card 404s. `skills/hunk-release/SKILL.md` gains the command beside the existing post-tag regeneration.

No new dependency — Playwright is already a website devDependency, and this follows `capture-media.ts`, including its `FFMPEG`-style env escape hatch (`CHROMIUM`) for pointing at another browser binary.

## One thing worth a look

`formatJson` in the generator emits JSON in the repository formatter's own shape, keeping a primitive array inline while it fits the print width. Without it, `bun run format` rewrote `cards.json` and `--check` reported it stale immediately after generating — the same class of conflict the YAML quoting fix solved in #783, and the reason `generate-docs.ts` pre-formats its Markdown tables. The 100-column threshold was measured against oxfmt rather than assumed, and is covered by tests including the boundary case.

## Verification

90 generator unit tests, 58 browser tests (2 new, asserting exactly one `og:image` per changelog page, that both images are actually served as `image/png`, and that docs pages keep the global image), plus `typecheck`, `lint`, `format:check`, `check:docs`, `check:changelog`, `website:build`, and `website:links`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01W4Q5TMBUvXD88aEyPHFb5c)_

### CODE REVIEW

File: website/scripts/generate-og.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+// A stale image for a series that no longer exists would keep being served; a full run owns the
+// directory, while a targeted run only touches what it was asked for.
+if (requested.size === 0) {
+  rmSync(outputDir, { recursive: true, force: true });
+  mkdirSync(outputDir, { recursive: true });
+}
+
+const browser = await chromium.launch(
+  process.env.CHROMIUM ? { executablePath: process.env.CHROMIUM } : {},
+);
```
[the remark this answers is not in the document]
Comment (benvinegar) [reply website/scripts/generate-og.ts]: Not applying this one — the cited `guidelines.mdc` lives in `modem-dev/modem`, and this repository has no validated environment-variable mechanism to route through. Every maintainer script here reads `process.env` directly:

- `website/scripts/capture-media.ts` — `process.env.FFMPEG ?? "ffmpeg"`
- `scripts/install-bin.ts` — `process.env.HUNK_INSTALL_DIR`
- `scripts/run-release-benchmark.ts` — `process.env.HUNK_RELEASE_BENCHMARK_SAMPLES`
- `scripts/launch-video/capture.ts` — `process.env.SCENES`

`CHROMIUM` was modelled deliberately on `capture-media.ts`'s `FFMPEG`: both are optional overrides pointing an opt-in maintainer script at a binary, in the same directory, and neither runs in CI or the website build. Introducing a validation layer for a single optional path would add the inconsistency the comment is trying to avoid.

Happy to revisit if a validated env helper lands in this repo.

---
_Generated by [Claude Code](https://claude.ai/code)_
